### PR TITLE
Add preset net

### PR DESCRIPTION
**Pull request by Steffy-zxf**
add preset net (bilstm, bow, cnn, dpcnn, gru, lstm) for text classification task, which was only available as ERNIE/RoBERTa/BERT/ELECTRA module

### CODE REVIEW

File: demo/text_classification/predict.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
 parser.add_argument("--max_seq_len", type=int, default=512, help="Number of words of the longest seqence.")
 parser.add_argument("--use_gpu", type=ast.literal_eval, default=False, help="Whether use GPU for finetuning, input should be True or False")
 parser.add_argument("--use_data_parallel", type=ast.literal_eval, default=False, help="Whether use data parallel.")
+parser.add_argument("--network", type=str, default='bilstm', help="Preset network which was connected after Transformer model, such as ERNIE, BERT ,RoBERTa and ELECTRA.")
```
Comment (ZeyuChen) [inline demo/text_classification/predict.py]: 预置网络，Predefined networks

File: paddlehub/finetune/task/classifier_task.py [hunk 2]
```diff
@@ -156,21 +161,73 @@ def _postprocessing(self, run_states):
...
+    It will use full-connect layer with softmax activation function to classify texts.
+    """
+
     def __init__(self,
-                 feature,
                  num_classes,
                  feed_list,
                  data_reader,
+                 token_feature=None,
+                 feature=None,
```
Comment (ZeyuChen) [inline paddlehub/finetune/task/classifier_task.py:173]: feature 要定义在在 token_feature前面比较合理。

File: paddlehub/finetune/task/classifier_task.py [hunk 3]
```diff
@@ -156,21 +161,73 @@ def _postprocessing(self, run_states):
...
                  config=None,
                  hidden_units=None,
                  metrics_choices="default"):
+        """
+        Args:
+            num_classes: total labels of the text classification task.
+            feed_list(list): the variable name that will be feeded to the main program
+            data_reader(object): data reader for the task. It must be one of ClassifyReader and LACClassifyReader.
+            token_feature(Variable): the feature will be used to connect the preset net. It must be the token-level feature, shape as [-1, seq_len, emb_size]. Default None.
+            feature(Variable): the feature will be used to classify texts. It must be the sentence-level feature, shape as [-1, emb_size]. Token_feature and feature couldn't be setted as the same time. One of them must be setted as not None. Default None.
```
Comment (ZeyuChen) [inline paddlehub/finetune/task/classifier_task.py]: Token_feature and feature couldn't be->
`token_feature` and `feature`
涉及参数变量需要用``括起来，避免歧义。

File: paddlehub/finetune/task/classifier_task.py [hunk 4]
```diff
@@ -179,10 +236,33 @@ def __init__(self,
...
+            self.seq_len = fluid.layers.data(
+                name="seq_len", shape=[1], dtype='int64', lod_level=0)
+            self.seq_len_used = fluid.layers.squeeze(self.seq_len, axes=[1])
+
+            # unpad the token_feature
+            unpad_feature = fluid.layers.sequence_unpad(
+                self.feature, length=self.seq_len_used)
+
+        if self.network:
+            # add preset net
```
Comment (ZeyuChen) [inline paddlehub/finetune/task/classifier_task.py]: pre-defined

File: paddlehub/finetune/task/classifier_task.py [hunk 5]
```diff
@@ -179,10 +236,33 @@ def __init__(self,
...
+                # deepcnn network is no need to unpad
+                cls_feats = net_func(
+                    self.feature, emb_dim=self.feature.shape[-1])
+            else:
+                cls_feats = net_func(unpad_feature)
+            logger.info(
+                "%s has been added in the TextClassifierTask!" % self.network)
+        else:
+            # not use preset net but to use fc net
+            cls_feats = fluid.layers.dropout(
```
Comment (ZeyuChen) [inline paddlehub/finetune/task/classifier_task.py:262]: This is dropout, where's the FC layer?

File: paddlehub/network/classification.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module provide nets for text classification
+"""
+
+from paddlehub.common.utils import version_compare
```
Comment (ZeyuChen) [inline paddlehub/network/classification.py]: It seems you can remove this import for simplify header

File: paddlehub/reader/nlp_reader.py [hunk 7]
```diff
@@ -293,28 +294,28 @@ def _pad_batch_records(self, batch_records, phase=None):
 
             return_list = [
```
Comment (ZeyuChen) [inline paddlehub/reader/nlp_reader.py]: `return_list`代码有点冗余。
是否可以通过if else的判断后，在原来return_list的基础上进行append，而不用每次都进行重新定义，这样代码也比较冗长。

**Review by ZeyuChen — CHANGES_REQUESTED** [review]
Please update your code according to my comments.

File: paddlehub/finetune/task/classifier_task.py (hunk 2 above)
Comment (Steffy-zxf) [reply paddlehub/finetune/task/classifier_task.py:173]: 已修改

File: demo/text_classification/predict.py (hunk 1 above)
Comment (Steffy-zxf) [reply demo/text_classification/predict.py]: > 预置网络，Predefined networks

已修改

File: paddlehub/finetune/task/classifier_task.py (hunk 2 above)
Comment (Steffy-zxf) [reply paddlehub/finetune/task/classifier_task.py:173]: > feature 要定义在在 token_feature前面比较合理。

已修改

File: paddlehub/finetune/task/classifier_task.py (hunk 3 above)
Comment (Steffy-zxf) [reply paddlehub/finetune/task/classifier_task.py]: > Token_feature and feature couldn't be->
> `token_feature` and `feature`
> 涉及参数变量需要用``括起来，避免歧义。

已修改

File: paddlehub/finetune/task/classifier_task.py (hunk 4 above)
Comment (Steffy-zxf) [reply paddlehub/finetune/task/classifier_task.py]: > pre-defined

done

File: paddlehub/finetune/task/classifier_task.py (hunk 5 above)
Comment (Steffy-zxf) [reply paddlehub/finetune/task/classifier_task.py:262]: > This is dropout, where's the FC layer?

line 267-280

File: paddlehub/network/classification.py (hunk 6 above)
Comment (Steffy-zxf) [reply paddlehub/network/classification.py]: > It seems you can remove this import for simplify header

done

File: paddlehub/reader/nlp_reader.py (hunk 7 above)
Comment (Steffy-zxf) [reply paddlehub/reader/nlp_reader.py]: > `return_list`代码有点冗余。
> 是否可以通过if else的判断后，在原来return_list的基础上进行append，而不用每次都进行重新定义，这样代码也比较冗长。

已修改

**Review by ZeyuChen — APPROVED** [review]
LGTM

**Pull request merged**